harpy/validation/fastq.py

```
@@ -53,7 +59,22 @@ def __init__(self, filenames):
             )
         
         self.count = len({re.sub(bn_r, "", i, flags = re.IGNORECASE) for i in uniqs})
-    
+
+        if detect_bc:
+            for i in range(min(6, self.count)):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

(B007)